mso/resource_mso_schema_template_anp_epg.go

```diff
@@ -510,6 +758,69 @@ func resourceMSOSchemaTemplateAnpEpgDelete(d *schema.ResourceData, m interface{}
 	}
+	if epg_type, ok := d.GetOk("epg_type"); ok {
+		epgType = epg_type.(string)
+	}
+
+	service := make(map[string]interface{})
+	if val, ok := d.GetOk("cloud_service_epg_config"); ok {
+		service_list := val.(*schema.Set).List()
+		for _, val := range service_list {
+
+			map_service := make(map[string]interface{})
+			inner_service := val.(map[string]interface{})
+
+			if inner_service["access_type"] == "private" {
+				map_service["accessType"] = "Private"
+			} else if inner_service["access_type"] == "public" {
+				map_service["accessType"] = "Public"
+			} else if inner_service["access_type"] == "public_and_private" {
+				map_service["accessType"] = "PublicAndPrivate"
+			}
+
+			if inner_service["deployment_type"] == "cloud_native" {
+				map_service["deploymentType"] = "CloudNative"
+			} else if inner_service["deployment_type"] == "cloud_native_managed" {
+				map_service["deploymentType"] = "CloudNativeManaged"
+			} else if inner_service["deployment_type"] == "third_party" {
+				map_service["deploymentType"] = "Third-party"
+			}
+
+			if inner_service["service_type"] == "azure_api_management_services" {
+				map_service["serviceType"] = "Azure-ApiManagement"
+			} else if inner_service["service_type"] == "azure_cosmos_db" {
+				map_service["serviceType"] = "Azure-CosmosDB"
+			} else if inner_service["service_type"] == "azure_databricks" {
+				map_service["serviceType"] = "Azure-Databricks"
+			} else if inner_service["service_type"] == "azure_sql" {
+				map_service["serviceType"] = "Azure-SqlServer"
+			} else if inner_service["service_type"] == "azure_storage" {
+				map_service["serviceType"] = "Azure-Storage"
+			} else if inner_service["service_type"] == "azure_storage_blob" {
+				map_service["serviceType"] = "Azure-StorageBlob"
+			} else if inner_service["service_type"] == "azure_storage_file" {
+				map_service["serviceType"] = "Azure-StorageFile"
+			} else if inner_service["service_type"] == "azure_storage_queue" {
+				map_service["serviceType"] = "Azure-StorageQueue"
+			} else if inner_service["service_type"] == "azure_storage_table" {
+				map_service["serviceType"] = "Azure-StorageTable"
+			} else if inner_service["service_type"] == "azure_kubernetes_services" {
+				map_service["serviceType"] = "Azure-AksCluster"
+			} else if inner_service["service_type"] == "azure_ad_domain_services" {
+				map_service["serviceType"] = "Azure-ADDS"
+			} else if inner_service["service_type"] == "azure_contain_registry" {
+				map_service["serviceType"] = "Azure-ContainerRegistry"
+			} else if inner_service["service_type"] == "azure_key_vault" {
+				map_service["serviceType"] = "Azure-KeyVault"
+			} else if inner_service["service_type"] == "redis_cache" {
+				map_service["serviceType"] = "Azure-Redis"
+			} else if inner_service["service_type"] == "custom" {
+				map_service["serviceType"] = "Custom"
+				map_service["customSvcType"] = inner_service["custom_service_type"]
+			}
+			service = map_service
```

Comment:
Also to avoid doing this 3 times we could use the same thing we did with backbone area and modify it in the user input section.